src/hotspot/os/linux/os_linux.cpp

```diff
@@ -6456,6 +6456,59 @@ void os::Linux::restore() {
   }
 }
 
+static bool is_fd_ignored(int fd, const char *path) {
+  int path_len = path ? strlen(path) : -1;
+  if (!strncmp("/proc/", path, 6) && !strcmp("/fd", path + path_len - 3)) {
+    // silently ignore us reading our own descriptors
+    return true;
+  }
```

Comment:
`is_fd_ignored` can be just not called for a fd from the dir (https://man7.org/linux/man-pages/man3/dirfd.3.html), that would save us from this check